tmt/base.py

```diff
@@ -1973,6 +1973,37 @@ def __init__(self,
         self._tree = tree
         self._custom_context = context
 
+    @classmethod
+    def grow(
+            cls,
+            *,
+            path: Path = Path.cwd(),
+            tree: Optional[fmf.Tree] = None,
+            context: Optional[tmt.utils.FmfContextType] = None,
+            logger: Optional[tmt.log.Logger] = None) -> 'Tree':
+        """
+        Initialize tmt tree from directory path or given fmf tree.
+
+        This method serves as an entry point for interactive use of
+        tmt-as-a-library, providing sane defaults.
+
+        .. warning::
+
+           This method has a **very** limited use case, i.e. to help
+           bootstrapping interactive tmt sessions. Using it anywhere
+           outside of this scope should be ruled out.
+        """
+
+        import tmt.plugins
+
+        tmt.plugins.explore()
```

Comment:
Anyway: there will be more plugins. Having them discovered and loaded in one place makes definitely sense to me. On the other hand, I'm not sure whether it might have unexpected side effects or not. I'd probably leave it for a dedicated patch, as it could affect things unrelated to the interactive entry point.

Reply:
Ack, let's cover that in a separate pull request.

Reply:
Proposing in https://github.com/teemtee/tmt/pull/1932